integration/client-cli/client-registration-cli/src/main/java/org/keycloak/client/registration/cli/util/IoUtil.java

```diff
@@ -81,43 +73,14 @@ public static void waitFor(long millis) {
         }
     }
 
-    public static String readSecret(String prompt, CommandInvocation invocation) {
-
-        // TODO Windows hack - masking not working on Windows
-        char maskChar = OS_ARCH.isWindows() ? 0 : '*';
-        ConsoleBuffer consoleBuffer = new AeshConsoleBufferBuilder()
-                .shell(invocation.getShell())
-                .prompt(new Prompt(prompt, maskChar))
-                .create();
-        InputProcessor inputProcessor = new AeshInputProcessorBuilder()
-                .consoleBuffer(consoleBuffer)
-                .create();
-
-        consoleBuffer.displayPrompt();
-
-        // activate stdin
-        Globals.stdin.setInputStream(System.in);
-
-        String result;
-        try {
-            do {
-                result = inputProcessor.parseOperation(invocation.getInput());
-            } while (result == null);
-        } catch (Exception e) {
-            throw new RuntimeException("^C", e);
-        }
-        /*
-        if (!Globals.stdin.isStdinAvailable()) {
-            try {
-                return readLine(new InputStreamReader(System.in));
-            } catch (IOException e) {
-                throw new RuntimeException("Standard input not available");
-            }
+    public static String readSecret(String prompt) {
+        Console cons;
+        char[] passwd;
+        if ((cons = System.console()) != null &&
+            (passwd = cons.readPassword("%s", prompt)) != null) {
+            return new String(passwd);
         }
-         */
-        // Windows hack - get rid of any \n
-        result = result.replaceAll("\\n", "");
-        return result;
+        throw new RuntimeException("Console is not active, or no password provided");
```

Comment:
nit - We discussed this in the PR for kcadm as it's not very user-friendly, but nothing that might block this PR.

Reply:
I should have mentioned this will be addressed by the follow-on pr that removes code duplication between the clients.

Reply:
Ahh, I see; works for me.